src/commands/importWebApp/importWebApp.ts

```diff
@@ -0,0 +1,321 @@
+    } else {
+        window.withProgress(
+            {
+                location: ProgressLocation.Notification,
+                title: localize("importWebApp", `Importing Web App '${webAppName}' to API Management service ${node.root.serviceName} ...`),
+                cancellable: false
+            },
+            // tslint:disable-next-line:no-non-null-assertion
+            async () => {
+                ext.outputChannel.appendLine(localize("importWebApp", "Importing Web App from wildcard..."));
+                const apiId = apiUtil.genApiId(apiName);
+                ext.outputChannel.appendLine(localize("importWebApp", "Creating new API..."));
+                const nApi = await constructApiFromWebApp(apiId, pickedWebApp, apiName);
+                await node!.createChild({ apiName, apiContract: nApi });
+                const serviceUrl = "https://".concat(nonNullOrEmptyValue(nonNullValue(pickedWebApp.hostNames)[0]));
+                const backendId = `WebApp_${apiUtil.displayNameToIdentifier(webAppName)}`;
+                ext.outputChannel.appendLine(localize("importWebApp", "Checking backend entity..."));
+                await setAppBackendEntity(node!, backendId, apiName, serviceUrl, resourceGroup, webAppName);
+                ext.outputChannel.appendLine(localize("importWebApp", "Creating policies..."));
+                await node!.root.client.apiPolicy.createOrUpdate(node!.root.resourceGroupName, node!.root.serviceName, apiName, {
+                    format: "rawxml",
+                    value: createImportXmlPolicy(backendId)
+                });
+                ext.outputChannel.appendLine(localize("importWebApp", "Create operations for API..."));
+                const operations = await getWildcardOperationsForApi(apiId, apiName);
+                for (const operation of operations) {
+                    await node!.root.client.apiOperation.createOrUpdate(node!.root.resourceGroupName, node!.root.serviceName, apiName, nonNullOrEmptyValue(operation.name), operation);
+                }
+                ext.outputChannel.appendLine(localize("importWebApp", "Import Web App succeed!"));
+            }
+        ).then(async () => {
+            // tslint:disable-next-line:no-non-null-assertion
+            await node!.refresh();
+            window.showInformationMessage(localize("importWebApp", `Imported Web App '${webAppName}' to API Management succesfully.`));
+        });
+    }
+}
+
+export async function getPickedWebApp(node: ApiTreeItem | ApisTreeItem, webAppType: string): Promise<Site> {
+    const client = azureClientUtil.getClient(node.root.credentials, node.root.subscriptionId, node.root.environment);
+    const allfunctionApps = await listWebApps(client, webAppType);
+    return await pickWebApp(allfunctionApps);
+}
+
+export async function listWebApps(client: WebSiteManagementClient, webAppType: string): Promise<Site[]> {
+    const allWebApps: WebAppCollection = await client.webApps.list();
```

Comment:
do you need paging here

Reply:
I don't think we need paging here, it's not direct mapi call, we don't have "nextlink" as well
![image](https://user-images.githubusercontent.com/29927264/75184032-8dda6600-56f8-11ea-9d88-3c36a1f54c39.png)

Reply:
Mapi is very specific to API Management it is not a ARM concept